main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1753,8 +1754,16 @@ object Weeder2 {
             loc = tree.loc)
           sctx.errors.add(error)
           Validation.Success(Expr.Error(error))
-        // Case: run expr with eff { handlers... }
-        case (expr, withs) => Validation.Success(Expr.TryWith(expr, withs, tree.loc))
+        // Case: run expr with ..
+        case (expr, withs) => Validation.Success(Expr.RunWith(expr, withs, tree.loc))
+      }
+    }
+
+    private def visitHandlerExpr(tree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
```

Comment:
Grammar or?

Reply:
what do you mean?

Reply:
ah right, Grammar in the weeder would be based on the syntax tree I guess? but i can add it to parser